Introduce a CI workflow to project

barrioskevin opened the pull request:
*Please complete the checklist so the group can quickly understand your PR.*

### Category
- [ ] **Feature**
- [ ] **Bug Fix**
- [X] **Testing**
- [ ] **Documentation**
- [X] **Other**

### Domain
- [ ] **Frontend**
- [ ] **Backend**
- [X] **Both**

### Description
This pull request is an attempt to introduce a CI workflow into our project. This will be required for design doc 2 and we can add in any missing pieces in the future. This pr provides the changes needed to make sure that our project builds and passes test on each merge.

### Related Issue(s)
Fixes issue #27 

### New Additions
`.github/workflows/workflow.yml` - The github actions workflow file that describes the job and steps run on each pr/merge.
`pytest.ini` - pytest configuration file to make it easier to run tests, just need to type pytest on main directory.

### Modifications
*previous modifications to the tests are ignored since the test we rewritten in #30*
`.gitignore` - added the coverage report directory that pytest will output just in case so it wont get uploaded to the repo.
`backend/requirements.txt` - added in flake8 to requirements. also included version number for packages.

### New Dependencies
No new dependencies

### Visuals
Screenshot of the CI pipeline successfully executing on my fork.
![image](https://github.com/user-attachments/assets/2ee53160-513c-4130-959b-ec45adfc4656)

### Architecture of Your Changes
No new modifications to the actual projects architecture. This change will pretty much only affect the process of creating pull requests and merging any pull requests to the main repository. It will run each of the steps described in the yaml file.

### Usage and Integration
The changes made interact with the process of creating and merging pull requests. You will only need to make sure that your PR successfully passes the workflow.

### Testing
I tested my workflow using my fork. I made sure that I was able to push a new change successfully. I guess this would fall under integration testing? See the visual above!

### Mentions
I think the whole group should be aware of this in order to be more careful when making PRs 
@misherwa @courtmcgee2197 @leed85 @thecatfour @nnguyen144

Commit:
started a workflow

Commit:
workflow test

Commit:
added node setup to workflow

Commit:
add comments to workflow

Commit:
removed my testing branches

thecatfour commented on .github/workflows/workflow.yml:
```diff
@@ -0,0 +1,48 @@
+      - name: Run Django Backend Tests
+        run: |
+          cd backend
+          python manage.py test
```
So we're using the manage.py test for the backend tests instead of pytest? Since we have pytest, pytest-cov, and pytest-django, we could use pytest to make a more detailed report for the backend tests.

barrioskevin replied:
good call, i'll reconfigure it to use pytest instead with a more detailed report. i'll come back to this when thats done.

Commit:
pytest config for backend testing

Commit:
adjusted workflow for pytest

Commit:
removed my testing branches from workflow

barrioskevin commented:
Hello, @thecatfour i have updated the workflow to now use pytest, i also added step for uploading coverage report as artifact like how we did on the ci lab.

thecatfour commented on .github/workflows/workflow.yml:
```diff
@@ -0,0 +1,53 @@
+          python -m pip install --upgrade pip
+          pip install -r backend/requirements.txt
+
+      # HERE WE CAN ADD IN STEPS FOR LINTING/SECURITY/AND FORMATTING
```
Will we decide on these steps in a meeting, or should we just discuss it on discord? The current version should be fine, but if we are going to add more, we should probably discuss it soon.

nnguyen144 commented:
I think our CI workflow looks good, however when we did the CI assignment, we utilized flake8 to check for syntax and code style issues to make our code appear neater and easier to read. I think we can implement flake8 here as each one of us will potentially have different ways we write our code, making it harder for an individual to read another person's code.

Commit:
use latest version of tests

Commit:
use latest tests

Commit:
added flake 8 to workflow.yml

Commit:
updated requirements, added flake8

Commit:
removed my fork branch from workflow.yml
